resources/packs/gcp/gcp.lr

```diff
@@ -556,43 +559,340 @@ private gcp.storage.bucket @defaults("id") {
+  // Internal ID
+  id string
+  // Depending on the value of retentionUnit, this is used to determine if a backup needs to be deleted
+  retainedBackups int
+  // The unit retainedBackups represents
+  retentionUnit string
+}
+
+// GCP Cloud SQL Instance Settings Deny Maintenance Period
+private gcp.project.sqlService.instance.settings.denyMaintenancePeriod @defaults("startDate endDate") {
+  // Internal ID
+  id string
+  // Deny maintenance period end date.
+  endDate string
+  // Deny maintenance period start date.
+  startDate string
+  // Time in UTC when the deny maintenance period starts and ends
+  time string
+}
+
+// GCP Cloud SQL Instance Settings Insights Configuration
+private gcp.project.sqlService.instance.settings.insightsConfig {
+  // Internal ID
+  id string
+  // Whether Query Insights feature is enabled
+  queryInsightsEnabled bool
+  // Number of query execution plans captured by Insights per minute for all queries combined
+  queryPlansPerMinute int
+  //  Maximum query length stored in bytes
+  queryStringLength int
+  // Whether Query Insights will record application tags from query when enabled.
+  recordApplicationTags bool
+  // Whether Query Insights will record client address when enabled.
+  recordClientAddress bool
+}
+
+// GCP Cloud SQL Instance Settings IP Configuration
+private gcp.project.sqlService.instance.settings.ipConfiguration {
+  // Internal ID
+  id string
+  // The name of the allocated IP range for the private IP Cloud SQL instance
+  allocatedIpRange string
+  // The list of external networks that are allowed to connect to the instance using the IP
+  authorizedNetworks []gcp.project.sqlService.instance.settings.ipConfiguration.aclEntry
+  // Whether the instance is assigned a public IP address
+  ipv4Enabled bool
+  // The resource link for the VPC network from which the Cloud SQL instance is accessible for private IP
+  privateNetwork string
+  // Whether SSL connections over IP are enforced or not
+  requireSsl bool
+}
+
+// GCP Cloud SQL Instance Settings IP Configuration Authorized Networks
+private gcp.project.sqlService.instance.settings.ipConfiguration.aclEntry {
+  // Internal ID
+  id string
+  // The time when this access control entry expires
+  expirationTime time
+  // A label to identify this entry
+  name string
+  // The allowlisted value for the access control list
+  value string
+}
+
+// GCP Cloud SQL Instance Settings Location Preferences
+private gcp.project.sqlService.instance.settings.locationPreference @defaults("zone") {
+  // Internal ID
+  id string
+  // The App Engine application to follow
+  followGaeApplication string
+  // The preferred Compute Engine zone for the secondary/failover
+  secondaryZone string
+  // The preferred Compute Engine zone for the primary instance
+  zone string
+}
+
+// GCP Cloud SQL Instance Settings Maintenance Window
+private gcp.project.sqlService.instance.settings.maintenanceWindow @defaults("day hour") {
```

Comment:
This feels like one we want to keep as a separate resource

Reply:
not sure what you mean here. Is it to not scope the resource under `gcp.project.sqlService.instance.settings` but to have it as `gcp.maintenanceWindow` or is it something else?

Reply:
I just meant nothing TODO :-)